$NetBSD: patch-ak,v 1.3 2009/03/14 11:00:22 apb Exp $

Changes from NetBSD xsrc/external/mit/xorg-server/dist/\
hw/xfree86/xaa/xaaInitAccel.c

----------------------------
revision 1.2
date: 2009/02/19 20:02:02;  author: macallan;  state: Exp;  lines: +13 -0
Actually use scanline image writes to upload images if available
----------------------------

--- hw/xfree86/xaa/xaaInitAccel.c.orig	2009-03-14 11:23:10.000000000 +0200
+++ hw/xfree86/xaa/xaaInitAccel.c
@@ -821,6 +821,19 @@ XAAInitAccel(ScreenPtr pScreen, XAAInfoR
 	infoRec->FillImageWriteRectsFlags = infoRec->ImageWriteFlags;     
     }
 
+    /**** FillScanlineImageWriteRects ****/
+
+    if(!infoRec->FillImageWriteRects) {
+        if(HaveScanlineImageWriteRect && 
+               (infoRec->ScanlineImageWriteFlags &
+                LEFT_EDGE_CLIPPING_NEGATIVE_X) &&
+               (infoRec->ScanlineImageWriteFlags & LEFT_EDGE_CLIPPING)) {
+           infoRec->FillImageWriteRects = XAAFillScanlineImageWriteRects;
+           infoRec->FillImageWriteRectsFlags = 
+               infoRec->ScanlineImageWriteFlags;
+       }
+    }
+
     /**** WriteBitmap ****/
 
     if(infoRec->WriteBitmap && 
